native/common/include/jp_env.h

```diff
@@ -101,19 +62,26 @@ namespace JPEnv
 
 	void attachJVM(const string& vmPath);
 
+	void detachCurrentThread();
 	void attachCurrentThread();
 	void attachCurrentThreadAsDaemon();
 	bool isThreadAttached();
+	void assertJVMRunning(const char* function, const char* file, int line);

```

Comment:
The assertJVMRunning is now used uniformally on the Python module interface to ensure that it will check any use of Java including edge cases such as JVM shutdown.